packages/daemon/src/daemon.js

```diff
@@ -1006,12 +1015,48 @@ const makeDaemonCore = async (
   };
 
   /** @type {import('./types.js').DaemonCore['incarnateGuest']} */
-  const incarnateGuest = async hostHandleFormulaIdentifier => {
-    const formulaNumber = await randomHex512();
-    const { formulaIdentifier: storeFormulaIdentifier } =
-      await incarnatePetStore();
-    const { formulaIdentifier: workerFormulaIdentifier } =
-      await incarnateWorker();
+  const incarnateGuest = async (hostFormulaIdentifier, hooks) => {
+    const {
+      guestFormulaNumber,
+      hostHandleFormulaIdentifier,
+      storeFormulaIdentifier,
+      workerFormulaIdentifier,
+    } = await formulaGraphMutex.enqueue(async () => {
+      const [
+        ownFormulaNumber,
+        hostHandle,
+        storeIncarnation,
+        workerIncarnation,
+      ] = await Promise.all([
+        randomHex512(),
+        randomHex512(),
+        randomHex512(),
+        randomHex512(),
+      ]).then(([own, handle, store, worker]) =>
+        Promise.all([
+          own,
+          incarnateNumberedHandle(handle, hostFormulaIdentifier),
+          incarnateNumberedPetStore(store),
+          incarnateNumberedWorker(worker),
```

Comment:
This parallelism is probably more elaborate than necessary given that the incarnations are going to be serialized.